src/charm.py

```diff
@@ -106,6 +114,32 @@ def on_config_changed_handler(event, fw_adapter):
         time.sleep(1)
 
 
+def on_new_am_rel_handler(event, fw_adapter):
+    if not fw_adapter.am_i_leader():
```

Comment:
What do you think btw about changing the `am_i_leader` signature to the `is_leader` or similar?

Reply:
I'm starting to think both don't clearly state enough that we're asking whether the current unit is the leader.

Reply:
What about `unit_is_leader` as a property?

Reply:
Sounds better - go for it!

Reply:
I've decided to use `unit_is_leader` but keep it a method (rather than convert it to a property) so as to make it somewhat consistent with the underlying framework method call.